### PR TITLE
Configured sass and css in webpack

### LINKED ISSUE #36: Configure webpack integration for SASS and CSS

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #36 

#### What's this PR do?
Sets up CSS and Sass integration with webpack, and points the index view to the right place. It also breaks the webpack javascript into three chunks: style, inner, and outer (feel free to suggest better names)

#### Where should the reviewer start?
webpack.config.dev.js

#### How should this be manually tested?
Go to the index page, view source. The javascript links should be there and should be valid links. Also, if you add style information to static/scss/layout.scss, it should immediately show up in your browser due to the hot reloading

### CODE REVIEW

**Comment by bdero** [comment]
This doesn't seem to be working on the deployed app:

File: ui/templates/base.html [hunk 1]
```diff
@@ -9,6 +10,7 @@
   <body>
     {% block content %}
     {% endblock %}
+  <script type="text/javascript" src="{{ outer_src }}"></script>
```
Comment (jamiefolsom) [inline ui/templates/base.html]: `outer`, as in non-dashboard?

Comment (noisecapella) [inline ui/templates/base.html]: Yes, I'm open to better names if you have one

File: ui/views.py [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    """
+    if settings.DEBUG and settings.USE_WEBPACK_DEV_SERVER:
+        host = request.get_host().split(":")[0]
+
+        return "{host_url}/{bundle}".format(
+            host_url=settings.WEBPACK_SERVER_URL.format(host=host),
+            bundle=bundle_name
+        )
+    else:
+        return static("bundles/{bundle}".format(bundle=bundle_name))
```
Comment (bdero) [inline ui/views.py]: Wasn't there a test associated with this code in teachersportal?

File: npm_install_if_prod.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+set -euf -o pipefail
+if [[ "$NODE_ENV" != "development" ]]
+then
+    node node_modules/webpack/bin/webpack.js --config webpack.config.prod.js
+fi
```
Comment (bdero) [inline npm_install_if_prod.sh]: :+1:

File: ui/views.py (hunk 2 above)
Comment (noisecapella) [inline ui/views.py]: Yes, I'll copy it over

File: ui/templates/base.html (hunk 1 above)
Comment (jamiefolsom) [inline ui/templates/base.html]: No that's fine, just want to understand. I have been thinking in terms of subdomains: www and app. Not sure others agree with that.

**Comment by bdero** [comment]
Verified it works on the deploy app now. Firing it up to try it out locally.

**Comment by bdero** [comment]
:+1:, I confirmed this is working locally for me. Although this will need a rebase since https://github.com/mitodl/micromasters/pull/28 created fresh conflicts.

File: ui/templates/base.html (hunk 1 above)
Comment (pdpinch) [inline ui/templates/base.html]: public? or anonymous_access?

(If I'm understanding the context correctly)

**Pull request merged**